diff --git a/src/java/org/apache/commons/lang/StringUtils.java b/src/java/org/apache/commons/lang/StringUtils.java
index 72b4a491..17f7d8cf 100644
--- a/src/java/org/apache/commons/lang/StringUtils.java
+++ b/src/java/org/apache/commons/lang/StringUtils.java
@@ -1045,7 +1045,13 @@ public class StringUtils {
         if (str == null || searchStr == null) {
             return false;
         }
-        return contains(str.toUpperCase(), searchStr.toUpperCase());
+        for (int i = 0; i <= str.length() - searchStr.length(); i++) {
+            if (str.regionMatches(true, i, searchStr, 0, searchStr.length())) {
+                return true;
+            }
+        }
+        return false;
+
     }
 
     // IndexOfAny chars
@@ -5764,4 +5770,4 @@ public class StringUtils {
         return str.regionMatches(ignoreCase, strOffset, suffix, 0, suffix.length());
     }
 
-}
+}
\ No newline at end of file
